clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -341,6 +341,14 @@ public void close() {
     }
 
     /**
+     * Flush any outgoing buffers and close the network client.
+     */
+    @Override
+    public void closeGracefully() {
+        this.selector.closeGracefully();
```

Comment:
The call to `closeGracefully` can result in some responses being returned from the brokers. Is it intentional that we do not invoke the completion handlers? I think this probably makes sense since anything awaiting the responses has probably shutdown by now, but wanted to check.